dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CollectionRestRepository.java

```diff
@@ -93,11 +108,26 @@ public CollectionRest findOne(Context context, UUID id) {
     @Override
     public Page<CollectionRest> findAll(Context context, Pageable pageable) {
         try {
-            long total = cs.countTotal(context);
-            List<Collection> collections = cs.findAll(context, pageable.getPageSize(),
-                Math.toIntExact(pageable.getOffset()));
-            return converter.toRestPage(collections, pageable, total, utils.obtainProjection());
-        } catch (SQLException e) {
+            if (authorizeService.isAdmin(context)) {
+                long total = cs.countTotal(context);
+                List<Collection> collections = cs.findAll(context, pageable.getPageSize(),
+                    Math.toIntExact(pageable.getOffset()));
+                return converter.toRestPage(collections, pageable, total, utils.obtainProjection());
+            } else {
+                List<Collection> collections = new LinkedList<Collection>();
+                DiscoverQuery discoverQuery = new DiscoverQuery();
```

Comment:
Could we add an inline comment here to describe the query we are running?  If I understand this correctly, we're looking for all Collections that the current user has access to?

Reply:
done here https://github.com/DSpace/DSpace/pull/2707/commits/9ce5c586269ebdde32e87d1d7591be15734f7dd3#diff-c7c44609573d540f6f0c0efa196ed6ffR118